OrbitLinuxTracing/PerfEventRingBuffer.cpp

```diff
@@ -121,8 +121,11 @@ void PerfEventRingBuffer::SkipRecord(const perf_event_header& header) {
   WriteRingBufferTail(metadata_page_, new_tail);
 }
 
-void PerfEventRingBuffer::ConsumeRecord(const perf_event_header& header, void* record) {
-  ReadAtTail(static_cast<uint8_t*>(record), header.size);
+void PerfEventRingBuffer::ConsumeRecord(const perf_event_header& header, void* record,
+                                        size_t size) {
+  // Sanity check so that we do not over-read from the ring-buffer and end up with random data.
+  CHECK(header.size >= size);
```

Comment:
The way `ConsumeRecord` is used, this should check for equality.

Reply:
yeah, this is what I started with :)

Reply:
Interesting - I think I did the check wrong yesterday. now it works fine. I will upload the version with more strict check.

Reply:
This check should be updated, or removed as it's already done by the public `ConsumeRecord`.

Reply:
updated